tools/common.py

```diff
@@ -0,0 +1,30 @@
+def func_dump_pipeline(pipeline, noKey=False):
```

Comment:
an example of what is dumped here would be nice to see in the comment for the function. Imnot going to hold that against this PR though. Can be a separate one